ModelVale/ModelVale/CustomClasses/CustomObjects/DatabaseObjects/ModelData.m

```diff
@@ -41,27 +41,29 @@ - (instancetype)initWithDictionary:(NSDictionary *)dict {
 }
 
 //MARK: Firebase
-+ (void) fetchFromReference: (FIRStorage*)storage docRef: (FIRDocumentReference*)docRef vc: (UIViewController*)vc completion:(void(^)(ModelData*))completion {
++ (void) fetchFromReference: (FIRDocumentReference*)docRef storage: (FIRStorage*)storage vc: (UIViewController*)vc completion:(void(^)(ModelData*))completion {
     [docRef getDocumentWithCompletion:^(FIRDocumentSnapshot * _Nullable snapshot, NSError * _Nullable error) {
         if(error != nil) {
             [vc presentError:@"Failed to fetch ModelData" message:error.localizedDescription error:error];
         }
         else {
             ModelData* d = [ModelData initWithImage:nil label:snapshot.data[@"label"] imagePath:snapshot.data[@"imagePath"]];
-            [d fetchAndSetImage:storage vc:vc];
-            completion(d);
+            [d fetchAndSetImage:storage vc:vc completion:^{
+                completion(d);
+            }];
         }
     }];
 }
 
-- (void) fetchAndSetImage: (FIRStorage*)storage vc: (UIViewController*)vc {
-    FIRStorageReference* imageRef = [[storage reference] child:self.imagePath];
-    [imageRef dataWithMaxSize:1 * 1024 * 1024 completion:^(NSData *data, NSError *error){
+- (void) fetchAndSetImage: (FIRStorage*)storage vc: (UIViewController*)vc completion:(void(^)(void))completion {
+    [[self getStorageRef:storage] dataWithMaxSize:10 * 4096 * 4096 completion:^(NSData *data, NSError *error){
         if (error != nil) {
             [vc presentError:@"Failed to download image" message:error.localizedDescription error:error];
```

Comment:
I know this is not  a change in this PR but I don't think it's a good practice for your model to be aware and store a reference of the view controller if you are following the MVC pattern so instead of sending the VC as a parameter in your completion you can return the status of the fetchAndSet, by returning an Enum or even just a Bool in your completion if it succeed then you can complete true or your succes enumParam an then when your view controller knows that it failed he is the one responsable of displaying his own errorUI. You can also achieve this with delegation, instead of storing the view controller you can tell the delegate(viewcontroller) that an error ocurred.